src/main/java/com/overops/plugins/sonar/rules/RuleDefinitionImplementation.java

```diff
@@ -0,0 +1,101 @@
+package com.overops.plugins.sonar.rules;
+
+import com.google.gson.Gson;
+import com.overops.plugins.sonar.measures.OverOpsMetrics;
+import com.overops.plugins.sonar.rules.checks.OverOpsChecks;
+import org.apache.commons.lang.StringUtils;
+import org.sonar.api.rule.RuleStatus;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.api.server.rule.RulesDefinitionAnnotationLoader;
+import org.sonar.api.utils.AnnotationUtils;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.plugins.java.api.CheckRegistrar;
+import org.sonar.plugins.java.api.JavaCheck;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.util.Collections;
+import java.util.stream.Collectors;
+
+public class RuleDefinitionImplementation implements RulesDefinition, CheckRegistrar {
+    private static final Logger LOGGER = Loggers.get(RuleDefinitionImplementation.class);
+    private static final String RESOURCE_BASE_PATH = "/org/sonar/l10n/rules";
+    private static final String OVER_OPS_REPOSITORY_NAME = "OverOps analyzer";
+    private static final String JAVA_LANGUAGE = "java";
+
+    @Override
+    public void register(RegistrarContext registrarContext) {
+        registrarContext.registerClassesForRepository(OverOpsChecks.REPOSITORY_KEY, OverOpsChecks.getChecks(), Collections.EMPTY_LIST);
+    }
+
+    @Override
+    public void define(Context context) {
+        NewRepository repository = context.createRepository(OverOpsChecks.REPOSITORY_KEY, JAVA_LANGUAGE).setName(OVER_OPS_REPOSITORY_NAME);
+        for (Class<? extends JavaCheck> check : OverOpsChecks.getChecks()) {
+            new RulesDefinitionAnnotationLoader().load(repository, new Class[]{check});
```

Comment:
perhaps we need to document or add some debug statement here when verbose debugging is turned on